Fix React Native Null Activity

Nightsd01 opened the pull request:
• In some cases, especially with projects using the react-native-navigation dependency, the initial Activity would be null, preventing the SDK from initializing.
• Added a check that will launch a re-attempt timer if this occurs.
• Updated build.gradle to use a newer version (26.0.2)
• Fixes #516, #547

Nightsd01 committed:
Fix React Native Null Activity

• In some cases, especially with projects using the react-native-navigation dependency, the initial Activity would be null, preventing the SDK from initializing.
• Added a check that will launch a re-attempt timer if this occurs.
• Updated build.gradle to use a newer version (26.0.2)

jkasten2 commented on android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java:
```diff
@@ -107,8 +106,23 @@ private JSONObject jsonFromErrorMessageString(String errorMessage) throws JSONEx
-      if (activity == null || oneSignalInitDone) {
-         Log.e("onesignal", "Unable to initialize the OneSignal SDK because activity is null " + (activity == null) + " or oneSignalInitDone" + oneSignalInitDone);
+      
+      if (activity == null) {
```
Not sure what values are on `mReactApplicationContext` but can we get an Android `Context` from this if `getCurrentActivity()` gives us `null`?

Nightsd01 replied:
@jkasten2 `getCurrentActivity()` is what is now used in RN to get the current activity. The old way of getting the activity (from the ReactApplicationContext object) is deprecated

https://github.com/facebook/react-native/releases/tag/v0.29.0

jkasten2 replied:
I believe `mReactApplicationContext.getApplicationContext()` should always be non-null.

Nightsd01 replied:
@jkasten2 `mReactApplicationContext.getApplicationContext()` behaves the same way. 

It seems that as of 2016, React-Native no longer guarantees this method will return a non null activity at the start of the app. So I think I'll just change it to get the activity from `mReactApplicationContext` and still check for null, and if it's null, re-check in 50ms. It appears this resolves a pretty common issue in RN applications.

Nightsd01 committed:
Get Activity from RN Context

• Changes how the SDK gets a reference to the current activity during initialization

Nightsd01 committed:
Fallback to Context Initialization

• Changed the SDK so that if, during initialization, the initial activity does not exist, the SDK will use the application context instead

Nightsd01 committed:
Cleanup

• Cleans up initialization of the SDK, since Activity is a subclass of Context, we can simplify the initialization method.

avishayil commented:
Actually I still encounter this. Not seeing my device on the users list and when using function to get the player id it returns `null`.

rgomezp commented:
@avishayil 
So this was never resolved? Came up recently in #797
